doc/N-API Basic Types.md

```diff
@@ -0,0 +1,373 @@
+# N-API Documentation
+
+## Env
+
+The opaque data structure containing the environment in which the request is being run. 
+
+The Env object is usually created and passed by the calling system.
+
+```
+Env(napi_env env);
+```
+
+- `[in] env`: The `napi_env` environment from which to construct the `Env` object.
+
+```
+operator napi_env() const;
+```
+
+Returns the `napi_env` opaque data structure representing the environment. 
+
+```
+Object Global() const;
+```
+
+Returns the `Object` representing JavaScript Global Object.
+
+```
+Value Undefined() const;
+```
+
+Returns the `Value` representing JavaScript Undefined Object.
+
+```
+Value Null() const;
+```
+
+Returns the `Value` representing JavaScript Null Object.
+
+```
+bool IsExceptionPending() const;
+```
+
+Returns a `bool` indicating if an exception is pending.
+
+```
+Error GetAndClearPendingException();
+```
+
+Returns an `Error` object representing the pending exception, if any.
+
+## Value
+
+A Value is the C++ manifestation of a JavaScript value.
+
+Value is a the base class upon which other JavaScipt values like Number, Boolean, String, and Object are based. 
+
+```
+Value();
```

Comment:
For each of the methods I would add a heading. In this example:

### Default Constructor

and we might want to add the `Added in: ` which is used in the standard Node doc as well.

We may want to include the Returns line, if if it is just `Returns void` (or whatever we think is the best way to say returns nothing)

See https://nodejs.org/dist/latest-v8.x/docs/api/n-api.html#n_api_napi_create_reference in the N-API doc as the example I looked at to come up with these suggestions.